client-ts/signalr/src/HttpConnection.ts

```diff
@@ -116,25 +118,50 @@ export class HttpConnection implements IConnection {
         }
     }
 
-    private createTransport(transport: TransportType | ITransport, availableTransports: string[]): ITransport {
-        if ((transport === null || transport === undefined) && availableTransports.length > 0) {
-            transport = TransportType[availableTransports[0]];
+    private createTransport(requestedTransport: TransportType | ITransport, availableEndpoints: IConnectionEndpoint[], requestedTransferFormat: TransferFormat): ITransport {
+        if (this.isITransport(requestedTransport)) {
+            this.logger.log(LogLevel.Trace, "Connection was provided an instance of ITransport, using that directly");
+            return requestedTransport;
+        }
+
+        for (const endpoint of availableEndpoints) {
+            const transport = this.resolveTransport(endpoint, requestedTransport, requestedTransferFormat);
+            if (transport) {
+                return this.constructTransport(transport);
+            }
         }
-        if (transport === TransportType.WebSockets && availableTransports.indexOf(TransportType[transport]) >= 0) {
+
+        throw new Error("Unable to initialize any of the available transports");
+    }
+
+    private constructTransport(transport: TransportType) {
+        if (transport === TransportType.WebSockets) {
             return new WebSocketTransport(this.options.accessTokenFactory, this.logger);
         }
-        if (transport === TransportType.ServerSentEvents && availableTransports.indexOf(TransportType[transport]) >= 0) {
+
+        if (transport === TransportType.ServerSentEvents) {
             return new ServerSentEventsTransport(this.httpClient, this.options.accessTokenFactory, this.logger);
         }
-        if (transport === TransportType.LongPolling && availableTransports.indexOf(TransportType[transport]) >= 0) {
+
+        if (transport === TransportType.LongPolling) {
```

Comment:
All the if tests here are on `transport`. Simplify code to use switch on `transport`?